pandas/core/internals/blocks.py

```diff
@@ -973,8 +973,8 @@ def setitem(self, indexer, value) -> Block:
 
         # length checking
         check_setitem_lengths(indexer, value, values)
-
-        value = extract_array(value, extract_numpy=True)
+        if self.dtype != _dtype_obj:
```

Comment:
Do we even need the `extract_array` call at all? I think the point of it is to explicitly convert a Series / Index to an object-dtype ndarray. If that's the case you are trying to avoid I don't think even need the condition here

Reply:
I think we need the `extract_array` call, otherwise, some tests in the `indexing` folder will not pass.

Reply:
Interesting - do you know what the failures are? Are they Index-related?

Reply:
definitely add a comment here pointing back to the PR/issue

might be np_can_hold_element assumes extract_array has been called, don't quote me on that.

Reply:
@xr-chen can you confirm what @jbrockmendel is saying? I'm not sure yet if this is worth doing as is or if we should try to solve the issue comprehensively

Reply:
@WillAyd I think he was right, he definitely knows `extract_array` and `np_can_hold_element` better than me. Do you have any plans on fixing it comprehensively?